tick/optim/solver/tests/svrg_test.py

```diff
@@ -54,8 +88,119 @@ def test_variance_reduction_setting(self):
+        """...Test in SVRG that dense and sparse code matches in all possible
+        settings
+        """
+        variance_reductions = ['last', 'rand']
+        rand_types = ['unif', 'perm']
+        seed = 123
+        tol = 0.
+        max_iter = 20
+        X, y = self.simu_linreg_data(interc=-1.)
+        d = X.shape[1]
+
+        proxs = [
+            ProxTV(strength=1e-2, range=(0, d)),
+            ProxElasticNet(strength=1e-2, ratio=0.9, range=(0, d)),
+            ProxEquality(range=(0, d))
+        ]
+
+        model_dense, model_spars = self.get_dense_and_sparse_linreg_model(X, y)
+        step = 1 / model_spars.get_lip_max()
+        for variance_reduction, rand_type, prox in product(variance_reductions,
+                                                           rand_types, proxs):
+            solver_sparse = SVRG(step=step, tol=tol, max_iter=max_iter,
+                                 verbose=False,
+                                 variance_reduction=variance_reduction,
+                                 rand_type=rand_type, seed=seed) \
+                .set_model(model_spars) \
+                .set_prox(prox)
+
+            solver_dense = SVRG(step=step, tol=tol, max_iter=max_iter,
+                                verbose=False,
+                                variance_reduction=variance_reduction,
+                                rand_type=rand_type, seed=seed) \
+                .set_model(model_dense) \
+                .set_prox(prox)
+
+            solver_sparse.solve()
+            solver_dense.solve()
+            np.testing.assert_array_almost_equal(solver_sparse.solution,
+                                                 solver_dense.solution, 7)
+
+        X, y = self.simu_linreg_data(interc=0.)
+        model_dense, model_spars \
+            = self.get_dense_and_sparse_linreg_model(X, y, fit_intercept=False)
+        step = 1 / model_spars.get_lip_max()
+        for variance_reduction, rand_type, prox in product(variance_reductions,
+                                                           rand_types, proxs):
+            solver_sparse = SVRG(step=step, tol=tol, max_iter=max_iter,
+                                 verbose=False,
+                                 variance_reduction=variance_reduction,
+                                 rand_type=rand_type, seed=seed) \
+                .set_model(model_spars) \
+                .set_prox(prox)
+
+            solver_dense = SVRG(step=step, tol=tol, max_iter=max_iter,
+                                verbose=False,
+                                variance_reduction=variance_reduction,
+                                rand_type=rand_type, seed=seed) \
+                .set_model(model_dense) \
+                .set_prox(prox)
+
+            solver_sparse.solve()
+            solver_dense.solve()
+            np.testing.assert_array_almost_equal(solver_sparse.solution,
+                                                 solver_dense.solution, 7)
```

Comment:
We should be able to duplicate less code in this example (loop on the intercept?)

Reply:
Do we really care about code duplication in unit tests ?